src/emeus-vfl-parser.c

```diff
@@ -213,6 +213,39 @@ has_view (VflParser *parser,
   return g_hash_table_lookup (parser->views_set, name) != NULL;
 }
 
+/* Valid attributes */
+static const struct {
+  int len;
+  const char *name;
+} valid_attributes[] = {
+  { 5, "width" },
+  { 6, "height" },
+  { 7, "centerX" },
+  { 7, "centerY" },
+  { 3, "top" },
+  { 6, "bottom" },
+  { 4, "left" },
+  { 5, "right" },
+  { 5, "start" },
+  { 3, "end" },
+  { 8, "baseline" }
+};
+
+static char *
+get_offset_to (const char *str,
+               const char *tokens)
+{
+  char *offset = NULL;
+
+  for (int i = 0; i < strlen (tokens); i++)
```

Comment:
Better, but now you're checking the length of a constant string in every iteration of the `for` loop. :-)

```
  int n_tokens = strlen (tokens);

  for (int i = 0; i < n_tokens; i++)
    {
```

Or, if you want to avoid the `strlen()` entirely:

```
  const char *t = tokens;
  for (; *t != '\0'; t++)
    {
      if ((offset = strchr (str, *t)) != NULL)
         …
```